akka-actor-typed-tests/src/test/scala/akka/actor/typed/internal/InterceptSpec.scala

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.internal
+
+import akka.actor.testkit.typed.scaladsl.{ ActorTestKit, TestProbe }
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.{ ActorContext, ActorRef, Behavior, BehaviorInterceptor, ReceiveTarget, Signal, SignalTarget, TypedAkkaSpecWithShutdown }
+import org.scalatest.WordSpecLike
+
+class InterceptSpec extends ActorTestKit with WordSpecLike with TypedAkkaSpecWithShutdown {
+
+  def snitchingInterceptor(probe: ActorRef[String]) = new BehaviorInterceptor[String, String] {
+    override def aroundReceive(ctx: ActorContext[String], msg: String, target: ReceiveTarget[String]): Behavior[String] = {
+      probe ! ("before " + msg)
+      val b = target(msg)
+      probe ! ("after " + msg)
+      b
+    }
+
+    override def aroundSignal(ctx: ActorContext[String], signal: Signal, target: SignalTarget[String]): Behavior[String] = {
+      target(signal)
+    }
+
+    // keeping the instance equality as "isSame" for these
+  }
+
+  "Intercept" should {
+
+    "intercept messages" in {
+      val probe = TestProbe[String]()
+      val interceptor = snitchingInterceptor(probe.ref)
+
+      val ref: ActorRef[String] = spawn(BehaviorImpl.intercept(
```

Comment:
why isn't this testing at the `scaladsl.Behaviors` level?

Reply:
The test is in `internal` so I just followed that to not have dependencies on the dsls/introduce dep-circels, but no strong opinion.

Reply:
since it's anyway public api we should have a test exercising that so might be better let this be *the* scaladsl test